### PR TITLE
Add machine pool autoscaling API

### PR DESCRIPTION
**Pull request by zgalor**

### CODE REVIEW

**Comment by zgalor** [comment]
@nimrodshn @tzvatot @igoihman PTAL

File: model/clusters_mgmt/v1/cluster_nodes_type.model [hunk 1]
```diff
@@ -26,8 +26,13 @@ struct ClusterNodes {
 	Infra Integer
 
 	// Number of compute nodes of the cluster.
+	// Compute and autoscale_compute cannot be used together.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/cluster_nodes_type.model:29]: `AutoscaleCompute`

File: model/clusters_mgmt/v1/cluster_nodes_type.model [hunk 2]
```diff
@@ -26,8 +26,13 @@ struct ClusterNodes {
 	Infra Integer
 
 	// Number of compute nodes of the cluster.
+	// Compute and autoscale_compute cannot be used together.    
 	Compute Integer
 
+	// Details for auto-scaling the compute machine pool.
+	// Compute and autoscale_compute cannot be used together.
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/cluster_nodes_type.model:33]: `AutoscaleCompute`

File: model/clusters_mgmt/v1/machine_pool_type.model [hunk 3]
```diff
@@ -16,18 +16,23 @@ limitations under the License.
 
 // Representation of a machine pool in a cluster.
 class MachinePool {
-	// ID used to identify the cluster that this machinepool is attached to.
-	link Cluster Cluster
+  // ID used to identify the cluster that this machinepool is attached to.
+  link Cluster Cluster
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/machine_pool_type.model:20]: There seems to be some misalignment in the formatting of indentation.
In the other file, there's 4 spaces leading to comment, and a tab for the fields.
Here' it's 2 spaces for both comment and field.

Comment (zgalor) [reply model/clusters_mgmt/v1/machine_pool_type.model:20]: Indeed. Not sure why this happened.
How should it be?

Comment (zgalor) [reply model/clusters_mgmt/v1/machine_pool_type.model]: Fixed. RePTAL

**Comment by tzvatot** [comment]
LGTM

**Pull request merged**